src/effects/effectslot.cpp

```diff
@@ -348,16 +348,21 @@ void EffectSlot::loadEffectInner(const EffectManifestPointer pManifest,
 
     m_pControlLoaded->forceSet(1.0);
 
-    if (m_pEffectsManager->isAdoptMetaknobSettingEnabled()) {
-        if (adoptMetaknobFromPreset) {
+    switch (m_pEffectsManager->adoptMetaknobSetting()) {
+    case AdoptMetaknobValue::KEEP_ALL:
+        slotEffectMetaParameter(m_pControlMetaParameter->get(), true);
+        break;
+    case AdoptMetaknobValue::KEEP_TOP:
+        if (loadingFromPreset) {
```

Comment:
```suggestion
        if (loadingFromChainPreset) {
```
In contrast to the preset (default state) of a single effect.